packages/dx-chart-core/src/plugins/zoom-pan/computeds.ts

```diff
@@ -174,16 +176,22 @@ export const detachEvents = (node: any, handlers: EventHandlers) => {
 
 /** @internal */
 export const getRect = (
-  interactionWithArguments, interactionWithValues, prevCoords, nextCoords, rootRef, offset,
+  interactionWithArguments: Interaction,
+  interactionWithValues: Interaction,
+  initial: Location,
+  current: Location,
+  pane: Pane,
 ) => {
-  const bbox = rootRef.current.getBoundingClientRect();
-  const isZoomArgument = interactionWithArguments === 'both' || interactionWithArguments === 'zoom';
-  const isZoomValue = interactionWithValues === 'both' || interactionWithValues === 'zoom';
-  const x = isZoomArgument ? Math.min(prevCoords[0], nextCoords[0]) : bbox.left - offset[0];
-  const width = isZoomArgument ? Math.abs(prevCoords[0] - nextCoords[0]) : bbox.width;
-  const y = isZoomValue ? Math.min(prevCoords[1], nextCoords[1]) : bbox.top - offset[1];
-  const height = isZoomValue ? Math.abs(prevCoords[1] - nextCoords[1]) : bbox.height;
+  const isZoomArgument = checkInteraction(interactionWithArguments, 'none', 'pan');
+  const isZoomValue = checkInteraction(interactionWithValues, 'none', 'pan');
+  const x = isZoomArgument ? Math.min(initial[0], current[0]) : 0;
+  const width = isZoomArgument ? Math.abs(initial[0] - current[0]) : pane.width;
+  const y = isZoomValue ? Math.min(initial[1], current[1]) : 0;
+  const height = isZoomValue ? Math.abs(initial[1] - current[1]) : pane.height;
   return {
     x, y, width, height,
   };
 };
+
+const checkInteraction = (interaction, value1, value2) =>
```

Comment:
There should be type definitions.
And it can be simplified - `(interaction, type) => interaction === 'both' || interaction === type`.

In `boundsForScale` interaction type is valid if it is `'both'` or matches `type`.
In `getRect` interaction type is valid if it is `'both'` or `'zoom'`.

Reply:
`(interaction, type) => interaction === 'both' || interaction === type` and `(interaction, type) => interaction !== 'both' && interaction !== type` - there is no critical differences and there is the same difficulty. It is really need to be changed?

Reply:
The point is to reduce number of arguments - from `type1` and `type2` to `type` (because `'both'` is always used). The meaning is - *Does specified interaction type match expected interaction type?*.
As for `(=== || ===)` or `(!== && !==)` - it does not matter.